spec/datadog/tracing/sampling/rule_sampler_spec.rb

```diff
@@ -77,6 +77,100 @@
     end
   end
 
+  describe 'parse' do
+    subject(:parse) { described_class.parse(rules.to_json, rate_limit, default_sample_rate) }
+    let(:rules) { [rule] }
+    let(:rate_limit) { nil }
+    let(:default_sample_rate) { nil }
+
+    let(:actual_rule) do
+      subject
+      expect(parse.rules).to have(1).item
+      parse.rules[0]
+    end
+
+    let(:actual_rules) do
+      subject
+      parse.rules
+    end
+
+    context 'with sample_rate' do
+      let(:rule) { { sample_rate: 0.1 } }
+
+      it 'parses as a match any' do
+        expect(actual_rule.matcher.name).to eq(Datadog::Tracing::Sampling::SimpleMatcher::MATCH_ALL)
+        expect(actual_rule.matcher.service).to eq(Datadog::Tracing::Sampling::SimpleMatcher::MATCH_ALL)
+        expect(actual_rule.sampler.sample_rate).to eq(0.1)
+      end
+
+      context 'and name' do
+        let(:rule) { super().merge(name: 'test-name') }
```

Comment:
```suggestion
        let(:rule) {{ sample_rate: 0.1, name: 'test-name' } }
```

The `super()` is a little cryptic. I assume that is calling the `rule`  method from the parent context, but I'm unsure if it is worth calling super to avoid typing `sample_rate: 0.1`.

Also, it might be that if the internal of RSpec changed our specs could break

Reply:
that sounds good!